Add MEGAsync and MSEdge Beta plugins + Repairs

giovcandido opened the pull request:
Hey!

I'd like to add a plugin to install MEGAsync from its official repo.

I tested the plugin myself and it works like a charm.

giovcandido commented:
I added an alternative msedge script as well. This way people can opt to download the beta version.

Please, consider accepting it.

asdil12 commented on opi/plugins/ms_edge_beta.py:
```diff
@@ -0,0 +1,22 @@
+			url = 'https://packages.microsoft.com/yumrepos/edge',
+			gpgkey = 'https://packages.microsoft.com/keys/microsoft.asc'
+		)
+
```
If you look at the existing `ms_edge.py`, you will find that there is some trick needed to prevent the rpm post install script from messing with the repo configuration by creating a new repo: https://github.com/openSUSE/opi/blob/master/opi/plugins/ms_edge.py#L22

I guess that it is likely that we need a similar trick here as well.

You can view the script by running `rpm -qi --scripts microsoft-edge-beta`.
Please make sure that it is covered as well.
In the script look for something like `DEFAULTS_FILE="/etc/default/microsoft-edge-dev"`.
If that file exists and is empty, the post install script should do nothing.

asdil12 replied:
Ah - here it is:
```
DEFAULTS_FILE="/etc/default/microsoft-edge-beta"
```

So you will need this here:
```suggestion

		# tell rpm post script not to mess with our repos
		subprocess.call(['sudo', 'rm', '-f', '/etc/default/microsoft-edge-beta'])
		subprocess.call(['sudo', 'touch', '/etc/default/microsoft-edge-beta'])
```

giovcandido replied:
I'm on it.

giovcandido replied:
It's done!

asdil12 commented on opi/plugins/megasync.py:
```diff
@@ -0,0 +1,39 @@
+
+		packages = ['megasync']
+
+		if which('nautilus') is not None:
```
This should be sufficient - also for the other ones:
```suggestion
		if which('nautilus'):
```

giovcandido replied:
Done!

asdil12 reviewed (changes requested):
looks fine overall - please see my comments on some details

giovcandido commented:
I notice brave post install script has the same behavior as msedge script. 

I'll upload a fix together.

giovcandido commented:
As I mentioned, I added the same fix to the brave script.

I renamed the existent msedge script to create a distinction between builds. When the stable version comes out, I can add another script.

Since the repo is the same, I made sure that the scripts wouldn't create two different repos.

asdil12 commented:
> I renamed the existent msedge script to create a distinction between builds. When the stable version comes out, I can add another script.

Seems that you forgot to delete the existing `ms_edge.py`.

asdil12 commented on opi/plugins/brave.py:
```diff
@@ -19,5 +19,9 @@ def run(cls, query):
+
 		opi.install_packages(['brave-browser'])
-		opi.ask_keep_repo('brave-browser')
+		opi.ask_keep_repo('brave-browser')
```
Please make sure that your files have a newline at the end of the file.
Your editor might not show the actual newline but eg vim complains with `[noeol]`.

giovcandido replied:
I changed my editor settings. It has the new line at the end now.

giovcandido commented:
> > I renamed the existent msedge script to create a distinction between builds. When the stable version comes out, I can add another script.
> 
> Seems that you forgot to delete the existing `ms_edge.py`.

Done.
